preprocess/datetime_map/mapper.py

```diff
@@ -0,0 +1,69 @@
+from datetime import datetime
+from enum import Enum
+
+import pandas as pd
+from dateutil.parser import parser
+
+from preprocessor import Preprocessor
+from report_data_d import _ColName
+
+
+class TimeOfDay(Enum):
```

Comment:
I think the correct practice is to create new columns for each of these times of days (one-hot encoding), e.g., a MORNING, AFTERNOON, EVENING, and NIGHT column. There's a link [here](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.OneHotEncoder.html) for sk-learn's encoder, but you can also just implement it yourself as well.